watson_developer_cloud/discovery_v1.py

```diff
@@ -12,620 +14,4824 @@
+        :param str natural_language_query: A natural language query that returns relevant documents by utilizing training data and natural language understanding. You cannot use `natural_language_query` and `query` at the same time.
+        :param str aggregation: An aggregation search uses combinations of filters and query search to return an exact answer. Aggregations are useful for building applications, because you can use them to build lists, tables, and time series. For a full list of possible aggregrations, see the Query reference.
+        :param int count: Number of documents to return.
+        :param list[str] return_fields: A comma separated list of the portion of the document hierarchy to return.
+        :param int offset: The number of query results to skip at the beginning. For example, if the total number of results that are returned is 10, and the offset is 8, it returns the last two results.
+        :param list[str] sort: A comma separated list of fields in the document to sort on. You can optionally specify a sort direction by prefixing the field with `-` for descending or `+` for ascending. Ascending is the default sort direction if no prefix is specified.
+        :param bool highlight: When true a highlight field is returned for each result which contains the fields that match the query with `<em></em>` tags around the matching query terms. Defaults to false.
+        :param bool deduplicate: When `true` and used with a Watson Discovery News collection, duplicate results (based on the contents of the `title` field) are removed. Duplicate comparison is limited to the current query only, `offset` is not considered. Defaults to `false`. This parameter is currently Beta functionality.
+        :param str deduplicate_field: When specified, duplicate results based on the field specified are removed from the returned results. Duplicate comparison is limited to the current query only, `offset` is not considered. This parameter is currently Beta functionality.
+        :return: A `dict` containing the `QueryResponse` response.
+        :rtype: dict
+        """
+        if environment_id is None:
+            raise ValueError('environment_id must be provided')
+        if collection_ids is None:
+            raise ValueError('collection_ids must be provided')
+        params = {
+            'version':
+            self.version,
+            'collection_ids':
+            ",".join(collection_ids)
+            if isinstance(collection_ids, list) else collection_ids,
+            'filter':
+            filter,
+            'query':
+            query,
+            'natural_language_query':
+            natural_language_query,
+            'aggregation':
+            aggregation,
+            'count':
+            count,
+            'return_fields':
+            ",".join(return_fields)
+            if isinstance(return_fields, list) else return_fields,
```

Comment:
This looks awkward, although it seems to work okay. I suspect `yapf` formatted the code this way.

I made a quick test file, and it works as expected:

```python
# test.py
x = ['1','2','3']
params = {
    'list':
    ','.join(x)
    if True else 'hello, world'
}
print(params['list'])
```

```bash
> python test.py
1,2,3
```

Reply:
I noticed the same thing.  It turns out that if _any_ line in the object constructor has to break at the `:`, then yapf breaks all the lines consistently at the `:`.  It looks awful but it works.